synapse/cortex.py

```diff
@@ -4391,7 +4391,13 @@ async def addHttpExtApi(self, adef):
         except Exception as e:
             mesg = f'Invalid path for Extended HTTP API - cannot compile regular expression for [{path}] : {e}'
             raise s_exc.BadArg(mesg=mesg) from None
-        adef['iden'] = s_common.guid()
+
+        if adef.get('iden') is None:
+            adef['iden'] = s_common.guid()
```

Comment:
`adef.setdefault('iden', s_common.guid())` ?

Reply:
We don't typically do that for setdefaults? At least not for guids (so we avoid a function call when possible).